src/test/java/tests/web/InventorySortingTest.java

```diff
@@ -1,90 +1,66 @@
-        LOGGER.info("Sorted product names (A to Z): {}", sortedNamesAZ);
-        Assert.assertTrue(isSortedAscending(sortedNamesAZ), "Products are sorted correctly by Name (A to Z)");
-    }
-
-    @Test
-    public void testInventorySortingByNameZA() {
-        LOGGER.info("Starting test: testInventorySorting");
-        login(driver, "standard_user", "secret_sauce");
-
-        InventoryPage inventoryPage = new InventoryPage(driver);
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
 
-        inventoryPage.selectSortingOption("Name (Z to A)");
-        List<String> sortedNamesZA = inventoryPage.getProductNames();
-        LOGGER.info("Sorted product names (Z to A): {}", sortedNamesZA);
-        Assert.assertTrue(isSortedDescending(sortedNamesZA), "Products are sorted correctly by Name (Z to A)");
-    }
+import pages.InventoryPage;
 
-    @Test
-    public void testInventorySortingByPriceLowHigh() {
-        LOGGER.info("Starting test: testInventorySorting");
-        login(driver, "standard_user", "secret_sauce");
+import enums.SortType;
 
-        InventoryPage inventoryPage = new InventoryPage(driver);
+public class InventorySortingTest extends BaseTest {
 
-        inventoryPage.selectSortingOption("Price (low to high)");
-        List<Double> sortedPricesLowToHigh = inventoryPage.getProductPrices();
-        LOGGER.info("Sorted product prices (Low to High): {}", sortedPricesLowToHigh);
-        Assert.assertTrue(isSortedAscending(sortedPricesLowToHigh), "Products are sorted correctly by Price (Low to High)");
+    @DataProvider(name = "sortingOptions")
+    public Object[][] sortingOptions() {
+        return new Object[][] {
+                { SortType.NAME_A_TO_Z },
+                { SortType.NAME_Z_TO_A },
+                { SortType.PRICE_LOW_TO_HIGH },
+                { SortType.PRICE_HIGH_TO_LOW }
+        };
     }
 
-    @Test
-    public void testInventorySortingByPriceHighLow(){
-        LOGGER.info("Starting test: testInventorySorting");
-        login(driver, "standard_user", "secret_sauce");
-
-        InventoryPage inventoryPage = new InventoryPage(driver);
-
-        inventoryPage.selectSortingOption("Price (high to low)");
-        List<Double> sortedPricesHighToLow = inventoryPage.getProductPrices();
-        LOGGER.info("Sorted product prices (High to Low): {}", sortedPricesHighToLow);
-        Assert.assertTrue(isSortedDescending(sortedPricesHighToLow), "Products are sorted correctly by Price (High to Low)");
+    @Test(dataProvider = "sortingOptions")
+    public void testInventorySorting(SortType sortType) {
+        SoftAssert softAssert = new SoftAssert();
+        InventoryPage inventoryPage = new InventoryPage(getDriver());
+
+        inventoryPage.selectSortingOption(sortType.getOption());
+
+        List<Double> sortedPrices;
+        List<String> sortedNames;
+
+        switch (sortType) {
```

Comment:
Keep the test layer readable; instead of using switch in the test method, move the logic to the service layer.